File: src/client/java/com/mod/client/compat/KeyBindingCompat.java
package com.mod.client.compat;

import net.fabricmc.fabric.api.client.keymapping.v1.KeyMappingHelper;
import net.minecraft.client.KeyMapping;
import net.minecraft.resources.Identifier;

public class KeyBindingCompat {

    public static KeyMapping register(String translationKey, int keyCode, String category) {
        KeyMapping mapping = new KeyMapping(
                translationKey,
                keyCode,
                new KeyMapping.Category(Identifier.withDefaultNamespace(category))
        );
        return KeyMappingHelper.registerKeyMapping(mapping);
    }

    private KeyBindingCompat() {}
}

File: src/client/kotlin/com/mod/client/compat/KeyBindingCompat.kt
package com.mod.client.compat

import net.fabricmc.fabric.api.client.keymapping.v1.KeyMappingHelper
import net.minecraft.client.KeyMapping
import net.minecraft.resources.Identifier

object KeyBindingCompat {
    @JvmStatic
    fun register(translationKey: String, keyCode: Int, category: String): KeyMapping {
        val mapping = KeyMapping(
            translationKey,
            keyCode,
            KeyMapping.Category(Identifier.withDefaultNamespace(category))
        )
        return KeyMappingHelper.registerKeyMapping(mapping)
    }
}
